### ISSUE TITLE
'No tracking on this page' on every facebook page

### ISSUE THREAD

**Issue by manoharprabhu** [d4t4x/data-selfie#2]
The code `$("#pagelet_bluebar a[data-testid='blue_bar_profile_link']")` returns nothing and data selfie fails to track the page.

I have only the ID "fb2k_pagelet_bluebar" on the DOM.

**Comment by 01fade** [comment]
Thank you submitting this!
What language in Facebook are you using?

**Comment by manoharprabhu** [comment]
English (US)

**Comment by 01fade** [comment]
Ok, that is weird. Would you know why your Facebook is different than most of other people's? 
I don't think I can really replicate this problem, right now.
Did you try making the selector change on your end and run an unpackaged extension?

**Comment by manoharprabhu** [comment]
I tried disabling all other extension and running in incognito mode. Didn't help.
Am still looking into why my DOM structure is very different.

Till then I am using the unpacked extension with hardcoded pic link, username and profile link and it is working great.

**Comment by 01fade** [comment]
Ok. Good that there is a way for you to at least use it!

Please let me know when you find the reason why your dom is different! I think there are a few people who also have the same problem. But again, I don't know yet I could test that.

**Comment by blackout314** [comment]
Same problem again, locale it_IT

**Comment by blackout314** [comment]
```
<div class="_4-u2 mbm _5jmm _5pat _5v3q _4-u8" data-fte="1" data-ftr="1" id="mall_post_10155001356614522:6:0" aria-labelledby="js_jm js_jn js_jo js_jp" data-insertion-position="1">

<div class="_3ccb" data-gt

```

content_looked.js row 183
`        var posts = window.global.feed.find("div._4-u2.mbm._5v3q._4-u8").children($("div._3ccb._4-u8"));
`

**Comment by blackout314** [comment]
https://github.com/d4t4x/data-selfie/pull/7
i've fixed it

**Comment by 01fade** [comment]
@blackout314 Thank you for your comments, can you further explain what you mean?
Are you addressing 2 different problems here?

I think @manoharprabhu's problem was that it would not track in the news feed. That happened to you, too? 
And then your pull request #7  is for supporting group and FB page pages. Those I did not include on purpose, because I haven't tested if I can be sure likes and comments can also be tracked on those - so to be more clear in usage, tracking should only work in the news feed and on no other page. Does that make sense?

**Comment by alarai** [comment]
Hello @228miles 

Just wanted to tell I'm in the same case as @manoharprabhu, I've just discovered the extension and was curious about it and it does not do any tracking after quite some time, there is no circle with number on the bottom left, and the extension information page just shows nothing.

I've also the same **fb2k_pagelet_bluebar** and no **pagelet_bluebar** 

On my side I'm using facebook in french. I'll try to see if I find how to use the unpacked extension.

**Comment by blackout314** [comment]
@228miles i've fixed the problem in PR, if you merge or try...

**Comment by 01fade** [comment]
@alarai Thank you for your comment? Yes, please test if you can. I think if more people report the id `#fb2k_pagelet_bluebar` that I can just add an OR statement, but if there even more ids out there, it gets complicated. 
Do you also have the id `#stream_pagelet`?

and thanks @blackout314 I have seen the PR, I think right now it's too fragile to also support groups and pages. But I investigate if just using #stream_pagelet would be sufficient. Also wondering if the user name and profile picture gets captured, too.

**Comment by alarai** [comment]
Hello @228miles 

I've tried to make it from source but it does not really seem to work more I've made a search in the src js and replaced **#pagelet_bluebar** by **#fb2k_pagelet_bluebar**. Then I installed the application but it seemed not too work either. Though that might have been me too because if I have the icon for Data Selfie in chrome, I cannot view the options or anything when installing manually.

I can confirm that I have a div with the id **stream_pagelet**

**Comment by japborst** [comment]
I have a `#pagelet_bluebar` in en_UK, but no tracking either... Not really sure how I can test what goes wrong

**Comment by 01fade** [comment]
@alarai did you follow all the instruction in the readme? So After you made the changes, you made a build. What do you mean by "I have the icon for Data Selfie in chrome, I cannot view the options or anything when installing manually"? Also I would recommend disabling the other Data Selfie extension from the Chrome Webstore, if you haven't.

@japborst I assume you know a little bit of code? The readme explains how to run the extension after you made changes. The changes should be made in the src folder. So in the specific case `src/js/content.js` all the way at the bottom, I am checking for `#pagelet_bluebar a[data-testid='blue_bar_profile_link']`. If that condition is true it should "Tracking on this page" in the console of your Facebook tab.

**Comment by alarai** [comment]
@228miles yeah I had been to quick trying to load directly the src in extension sorry. So I did try again, downloaded node.js to get the latest version and download every requirements to make it work. So finally I got a built version. But it's still not working. 

while I can make it work just a bit with **#fb2k_pagelet_bluebar** but then there is no real tracking the **#fb2k_pagelet_bluebar a[data-testid='blue_bar_profile_link']** remains false 

So i've searched quite more and found that my profile link is the following code : 
**<a accesskey="2" class="_5yf _4qx2" data-gt="{"chrome_nav_item":"timeline_chrome"}" href="https://www.facebook.com/myname">**

Meaning there is no blue_bar_profile_link either. 

So I've used chrome and made an ugly version with **#u_0_6 > div > div > div._2btk > div > div._9t5._398f > div._4qx3 > div > a** and it works, but it is really ugly. 

If there is anything that could be provided to help you do not hesitate to let me know, in the meanwhile I'll start with this :)

**Comment by 01fade** [comment]
@alarai ok sounds good. With an id like `#u_0_6` I feel a little suspicious that it won't work all the time. But keep me updated, please. If it works well, maybe I will add that as an if statement (in case `#pagelet_bluebar` is not present).

**Comment by alarai** [comment]
@228miles I'll let you know no problem. Actyually I did not really searched a lot just got the google inspector Selector tool to make it fast and try it. If I find something better I will let you know. I'll have to look after this one too I think **#pagelet_bluebar a[title='Profile']** as it probably don't work either.

**Comment by alarai** [comment]
Hello @228miles 

Just to keep you informed, right now I'm still using the code I've talked about the last time, and so far so good. The extension works like a charm. Only thing is I've not found any replacement for the **#pagelet_bluebar a[title='Profile']** but its difficult because I don't have a page that actually have this tag as a refrence so I could see what it is and find the equivalent on mine.

**Comment by osopolar** [comment]
In my case I have `#pagelet_bluebar`, but not `a[data-testid='blue_bar_profile_link']`. The profile link is:

```
<div class="_1k67 _4q39" data-click="profile_icon">
  <a class="_2s25" accesskey="2" data-gt="{&quot;chrome_nav_item&quot;:&quot;timeline_chrome&quot;}" href="https://www.facebook.com/my-profile-name" title="Profil">
    ...
  </a>
</div>
```

Language: German.

**Edit:**

Changing langauge to English didn't change anything, maybe it isn't about language but the location (in my case Argentina). Would the selector `#pagelet_bluebar div[data-click='profile_icon'] a` be helpful`?

**Comment by 01fade** [comment]
Do people have an opinion/experience on this current selector that I am using ```#pagelet_bluebar a._2s25._606w```?

**Comment by 01fade** [comment]
For anybody who comes here to find a fix for why Data Selfie doesn't seem to track please consider contributing by sending a screenshot of your code (of course black out any information you don't want to share). I have uploaded a video where I show how to get the view of it here: https://vimeo.com/234004265

You can post it here (preferred) or send me an email at support@dataselfie.it with the subject line of "[blue bar]".

Hopefully there will be enough submissions to support ```#pagelet_bluebar``` and  ```fb2k_pagelet_bluebar``` and also find out how to select the name/profile icon to verify that a user is logged in which then triggers the start of tracking.

P.S. quite some people also reported that just waited a few days and then it worked all of the sudden - I don't have any explanation for that, but just wanted to share this info.

**Comment by DarioCorno** [comment]
Mine shows #pagelet_bluebar but the counter sometimes (actually a lot) fails to start when I scroll. The console says "no newsfeed posts in view".

**Comment by 01fade** [comment]
@DarioCorno So, you are saying the counter works when you stop scrolling though? (if you then that's how it supposed to work. I don't want to count seconds on posts that you basically just scroll by, but once you stop because you are looking at something it should count)
